packages/backend/api/ipfs-pinning/snapshot-settings.ts

```diff
@@ -42,57 +35,35 @@ export default async (request: VercelRequest, response: VercelResponse) => {
     snapshot.verifyNewSnapshotSettings(originalSpaceSettings, snapshotSpaceSettings)
 
     // upload to IPFS and pinning
-    const ipfs = create({
-      host: "ipfs.infura.io",
-      port: 5001,
-      protocol: "https",
-      apiPath: "/api/v0",
+    const pinataResponds = await fetch(PINATA_BASE_URL + "/pinning/pinJSONToIPFS", {
+      method: "POST",
+
       headers: {
-        authorization: IPFS_INFURA_AUTH,
+        pinata_api_key: PINATA_API_KEY,
+        pinata_secret_api_key: PINATA_SECRET_API_KEY,
+        "Content-Type": "application/json",
       },
-    })
-    const { cid: rawCid } = await ipfs.add(JSON.stringify(snapshotSpaceSettings))
-    console.log("Respond CID from IPFS (Infura) raw:", rawCid)
-
-    const cid = rawCid.toV1().toString()
-    console.log("Converted CIDv1:", cid)
-
-    try {
-      const pinningRes = await fetch(`${PINNING_SERVICES_API_URL}/pins`, {
-        method: "POST",
-        body: JSON.stringify({
-          cid: cid,
-          name: `SnapshotSpaceSettings-${snapshotSpaceEnsName}`,
-        }),
-        headers: {
-          "Content-Type": "application/json",
-          Authorization: `Bearer ${PINNINGS_SERVICE_KEY}`,
-        },
-      })
-      console.log("Responds from Pinning Service API", pinningRes)
-    } catch (error) {
-      if (error.reason !== "DUPLICATE_OBJECT") {
-        return response
-          .status(500)
-          .setHeader("content-type", "application/json;charset=UTF-8")
-          .setHeader("Access-Control-Allow-Origin", "*")
-          .send("error from pinning service")
-      }
-      console.log("Pinning Service API error:", error)
-    }
+      body: JSON.stringify(snapshotSpaceSettings),
+    }).then((res) => res.json())
 
-    console.log("SUCCESS: Pinnings successfully. CIDv1:", cid)
+    console.log("Pinata responds", pinataResponds)
+    const { IpfsHash: cidV0 } = pinataResponds
     return response
       .status(200)
       .setHeader("content-type", "application/json;charset=UTF-8")
       .setHeader("Access-Control-Allow-Origin", "*")
       .send(
         JSON.stringify({
-          cidV1: cid,
+          cidV0,
         }),
       )
   } catch (e) {
     console.error(e)
-    return response.status(500).send("error")
+
+    return response
+      .status(500)
+      .setHeader("content-type", "application/json;charset=UTF-8")
+      .setHeader("Access-Control-Allow-Origin", "*")
+      .send("error")
```

Comment:
It'd be nice to surface the error message in these server responses

Reply:
I thought of it as a security precaution since the Pinata API might assume that it's returning an error to a trusted environment. Therefore we would have to inspect the logs for the actual error.

But I think you are right; this is not critical stuff, so we should probably just return it.

Thank you